cli/src/main/kotlin/commands/ReporterCommand.kt

```diff
@@ -252,43 +254,41 @@ class ReporterCommand : CliktCommand(
             reportSpecificOptionsMap[option.first] = option.second
         }
 
-        var failure = false
-        val reportDurationMap = mutableMapOf<Reporter, TimedValue<List<File>>>()
+        val reportDurationMap = measureTimedValue {
+            runBlocking(Dispatchers.Default) {
+                reportFormats.map { reporter ->
+                    async {
+                        val threadName = Thread.currentThread().name
+                        println("Generating the '${reporter.reporterName}' report in thread '$threadName'...")
+
+                        reporter to measureTimedValue {
+                            val options = reportOptionsMap[reporter.reporterName.toUpperCase()].orEmpty()
+                            runCatching { reporter.generateReport(input, outputDir, options) }
+                        }
+                    }
+                }.awaitAll()
+            }
+        }
 
-        reportFormats.forEach { reporter ->
-            val options = reportOptionsMap[reporter.reporterName.toUpperCase()].orEmpty()
+        var failureCount = 0
```

Comment:
As an alternative to incrementing a counter in a `forEach` block, I think it would be possible - and more functional style - to use the [partition()](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/partition.html) function to separate the successful from the failed reports. This yields two lists with failed and successful reports, which could be processed separately.

Reply:
I rewrote the code locally to see how it looks like, but I didn't like it as it resulted in a bit of code duplication in the separate loops for processing successes and failures.